livedns/domain.go

```diff
@@ -25,63 +27,63 @@ type SigningKey struct {
 // GetDomain returns a domain
-func (g *Gandi) GetDomain(fqdn string) (domain Domain, err error) {
-	_, err = g.askGandi(mGET, "domains/"+fqdn, nil, &domain)
+func (g *LiveDNS) GetDomain(fqdn string) (domain Domain, err error) {
+	_, err = g.client.Get("domains/"+fqdn, nil, &domain)
 	return
 }
 
 // ChangeAssociatedZone changes the zone associated to a domain
-func (g *Gandi) ChangeAssociatedZone(fqdn, uuid string) (response StandardResponse, err error) {
-	_, err = g.askGandi(mPATCH, "domains/"+fqdn, Domain{ZoneUUID: uuid}, &response)
+func (g *LiveDNS) ChangeAssociatedZone(fqdn, uuid string) (response client.StandardResponse, err error) {
+	_, err = g.client.Patch("domains/"+fqdn, Domain{ZoneUUID: uuid}, &response)
 	return
 }
 
 // DetachDomain detaches a domain from the zone it is attached to
-func (g *Gandi) DetachDomain(fqdn string) (err error) {
-	_, err = g.askGandi(mDELETE, "domains/"+fqdn, nil, nil)
+func (g *LiveDNS) DetachDomain(fqdn string) (err error) {
+	_, err = g.client.Delete("domains/"+fqdn, nil, nil)
 	return
 }
 
 // SignDomain creates a DNSKEY and asks Gandi servers to automatically sign the domain
-func (g *Gandi) SignDomain(fqdn string) (response StandardResponse, err error) {
+func (g *LiveDNS) SignDomain(fqdn string) (response client.StandardResponse, err error) {
 	f := SigningKey{Flags: 257}
-	_, err = g.askGandi(mPOST, "domains/"+fqdn+"/keys", f, &response)
+	_, err = g.client.Post("domains/"+fqdn+"/keys", f, &response)
 	return
 }
 
 // GetDomainKeys returns data about the signing keys created for a domain
-func (g *Gandi) GetDomainKeys(fqdn string) (keys []SigningKey, err error) {
-	_, err = g.askGandi(mGET, "domains/"+fqdn+"/keys", nil, &keys)
+func (g *LiveDNS) GetDomainKeys(fqdn string) (keys []SigningKey, err error) {
+	_, err = g.client.Get("domains/"+fqdn+"/keys", nil, &keys)
 	return
 }
 
 // DeleteDomainKey deletes a signing key from a domain
-func (g *Gandi) DeleteDomainKey(fqdn, uuid string) (err error) {
-	_, err = g.askGandi(mDELETE, "domains/"+fqdn+"/keys/"+uuid, nil, nil)
+func (g *LiveDNS) DeleteDomainKey(fqdn, uuid string) (err error) {
+	_, err = g.client.Delete("domains/"+fqdn+"/keys/"+uuid, nil, nil)
 	return
 }
 
 // UpdateDomainKey updates a signing key for a domain (only the deleted status, actually...)
-func (g *Gandi) UpdateDomainKey(fqdn, uuid string, deleted bool) (err error) {
-	_, err = g.askGandi(mPUT, "domains/"+fqdn+"/keys/"+uuid, SigningKey{Deleted: &deleted}, nil)
+func (g *LiveDNS) UpdateDomainKey(fqdn, uuid string, deleted bool) (err error) {
+	_, err = g.client.Put("domains/"+fqdn+"/keys/"+uuid, SigningKey{Deleted: &deleted}, nil)
 	return
 }
 
 // GetDomainNS returns the list of the nameservers for a domain
-func (g *Gandi) GetDomainNS(fqdn string) (ns []string, err error) {
-	_, err = g.askGandiFromBytes(mGET, "nameservers/"+fqdn, nil, &ns)
+func (g *LiveDNS) GetDomainNS(fqdn string) (ns []string, err error) {
+	_, err = g.client.Get("nameservers/"+fqdn, nil, &ns)
 	return
 }
```

Comment:
Please add:

```
// Nameservers represents a list of nameservers
type Nameservers struct {
       Nameservers []string `json:"nameservers,omitempty"`
}

// UpdateDomainNS returns the list of the nameservers for a domain
func (g *LiveDNS) UpdateDomainNS(fqdn string, ns []string) (err error) {
	_, err = g.client.Put("domain/domains/"+fqdn+"/nameservers", Nameservers{ns}, nil)
	return
}
```

(Obviously put the `type Nameservers struct` at the top of the file`.)

Reply:
This is a method on the Domains API not the LiveDNS API (https://api.gandi.net/docs/domains/#put-v5-domain-domains-domain-nameservers), so I'll add it there as I do that work.